righto.js

```diff
@@ -1,31 +1,30 @@
 'use strict';
 
-const fs = require('fs');
-const path = require('path');
+const fs            = require('fs');
 
-const righto = require('righto');
+const righto        = require('righto');
+const S             = require('sanctuary');
 
+const exit0         = require('./common/exit0');
+const exit1         = require('./common/exit1');
+const join          = require('./common/join');
 
-//    readFile :: String -> String -> Righto String
-const readFile = dir => file =>
-  righto(fs.readFile, path.join(dir, file), {encoding: 'utf8'});
+
+//    readFile :: String -> Righto String
+const readFile = S.curry3(righto, fs.readFile, S.__, {encoding: 'utf8'});
+
+//    concatFiles :: (String -> String) -> Righto String
+const concatFiles = path =>
+  righto.sync(S.joinWith(''),
+              righto.all(righto.sync(S.pipe([S.lines,
+                                             S.map(path),
+                                             S.map(readFile)]),
+                                     readFile(path('index.txt')))));
 
 
 const main = () => {
-  const dir = process.argv[2];
-
-  const file = readFile(dir)('index.txt');
-  const files = righto.sync(index => index.match(/^.*(?=\n)/gm).map(readFile(dir)), file);
-  const concatedFiles = righto.sync(results => results.join(''), righto.all(files));
-
-  concatedFiles((err, data) => {
-    if (err != null) {
-      process.stderr.write(String(err) + '\n');
-      process.exit(1);
-    } else {
-      process.stdout.write(data);
-      process.exit(0);
-    }
+  concatFiles(join(process.argv[2]))((err, data) => {
+    if (err == null) exit0(data); else exit1(err);
   });
 };
 
```

Comment:
The change to remove variables for each logical step goes violently against the grain of righto, and makes this, to my eyes, unreadable.

Also the change to use sanctuary within this example in my opinion muddies it, since a new viewer would have to look up how it works AND how righto works.

At this point, since all the examples use sanctuary, I don't see this as a useful comparison, but a contrived example that is designed explicitly to show of one or two particular examples.